### PR TITLE
@types/node: extend AbortSignal from EventTarget

### PR DESCRIPTION
**Pull request by no0dles**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/all.html#globals_class_abortsignal
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/node/v15/globals.d.ts [hunk 1]
```diff
@@ -337,6 +337,9 @@ interface AbortSignal {
      * Invoking this method will set this object's AbortSignal's aborted flag and signal to any observers that the associated activity is to be aborted.
      */
     abort(): void;
+
+    onabort: ((this: AbortSignal, event: Event) => any) | null;
+    addEventListener(type: 'abort', listener: (this: AbortSignal, event: Event) => any, options?: { once?: boolean }): void;
```
Comment (SimonSchick) [inline types/node/v15/globals.d.ts:342]: We are about to remove v15 types, pls revert these changes.

File: types/node/globals.d.ts [hunk 2]
```diff
@@ -62,6 +62,9 @@ interface AbortSignal {
      * Returns true if this AbortSignal's AbortController has signaled to abort, and false otherwise.
      */
     readonly aborted: boolean;
+
+    onabort: ((this: AbortSignal, event: Event) => any) | null;
```
Comment (SimonSchick) [inline types/node/globals.d.ts:66]: Can we change `any` to `void`?

File: types/node/globals.d.ts [hunk 3]
```diff
@@ -62,6 +62,9 @@ interface AbortSignal {
...
+    addEventListener(type: 'abort', listener: (this: AbortSignal, event: Event) => any, options?: { once?: boolean }): void;
```
Comment (SimonSchick) [inline types/node/globals.d.ts:67]: We should probably move the `once` to a shared place so it can be re-used, probably `events.d.ts`.

File: types/node/globals.d.ts (hunk 2 above)
Comment (SimonSchick) [reply types/node/globals.d.ts:66]: Also the `Event` class/interface is not yet defined in node typings.

**Review by SimonSchick — CHANGES_REQUESTED** [review]

File: types/node/globals.d.ts (hunk 2 above)
Comment (no0dles) [reply types/node/globals.d.ts:66]: Screenshot 2021-08-21 at 18 21 10

I picked any, so its compatible with lib.dom definition, is it save to use void?

File: types/node/globals.d.ts (hunk 3 above)
Comment (no0dles) [reply types/node/globals.d.ts:67]: sure, can i add an events namespace import to the globals.d.ts or is that an issue?
